core/error.js

```diff
@@ -0,0 +1,230 @@
+        if (evalOrigin == null) {
+          throw new Error("assert evalOrigin");
+        }
+        result += `${evalOrigin}, `;
+      }
+      result += "<anonymous>";
+    }
+
+    const lineNumber = callSite.getLineNumber();
+    if (lineNumber != null) {
+      result += `:${lineNumber}`;
+
+      const columnNumber = callSite.getColumnNumber();
+      if (columnNumber != null) {
+        result += `:${columnNumber}`;
+      }
+    }
+
+    return result;
+  }
+
+  // Keep in sync with `cli/fmt_errors.rs`.
+  function formatCallSite(callSite) {
+    let result = "";
+    const functionName = callSite.getFunctionName();
+
+    const isTopLevel = callSite.isToplevel();
+    const isAsync = callSite.isAsync();
+    const isPromiseAll = callSite.isPromiseAll();
+    const isConstructor = callSite.isConstructor();
+    const isMethodCall = !(isTopLevel || isConstructor);
+
+    if (isAsync) {
+      result += "async ";
+    }
+    if (isPromiseAll) {
+      result += `Promise.all (index ${callSite.getPromiseIndex()})`;
+      return result;
+    }
+    if (isMethodCall) {
+      const typeName = callSite.getTypeName();
+      const methodName = callSite.getMethodName();
+
+      if (functionName) {
+        if (typeName) {
+          if (!functionName.startsWith(typeName)) {
+            result += `${typeName}.`;
+          }
+        }
+        result += functionName;
+        if (methodName) {
+          if (!functionName.endsWith(methodName)) {
+            result += ` [as ${methodName}]`;
+          }
+        }
+      } else {
+        if (typeName) {
+          result += `${typeName}.`;
+        }
+        if (methodName) {
+          result += methodName;
+        } else {
+          result += "<anonymous>";
+        }
+      }
+    } else if (isConstructor) {
+      result += "new ";
+      if (functionName) {
+        result += functionName;
+      } else {
+        result += "<anonymous>";
+      }
+    } else if (functionName) {
+      result += functionName;
+    } else {
+      result += formatLocation(callSite);
+      return result;
+    }
+
+    result += ` (${formatLocation(callSite)})`;
+    return result;
+  }
+
+  function evaluateCallSite(callSite) {
+    return {
+      this: callSite.getThis(),
+      typeName: callSite.getTypeName(),
+      function: callSite.getFunction(),
+      functionName: callSite.getFunctionName(),
+      methodName: callSite.getMethodName(),
+      fileName: callSite.getFileName(),
+      lineNumber: callSite.getLineNumber(),
+      columnNumber: callSite.getColumnNumber(),
+      evalOrigin: callSite.getEvalOrigin(),
+      isToplevel: callSite.isToplevel(),
+      isEval: callSite.isEval(),
+      isNative: callSite.isNative(),
+      isConstructor: callSite.isConstructor(),
+      isAsync: callSite.isAsync(),
+      isPromiseAll: callSite.isPromiseAll(),
+      promiseIndex: callSite.getPromiseIndex(),
+    };
+  }
+
+  function createPrepareStackTrace(sourceMappingFn) {
```

Comment:
Can you add some documentation here about the type of the `sourceMappingFn`?

Reply:
Sure thing